sacro-app/main.js

```diff
@@ -31,16 +36,32 @@ function findAppPath() {
 }
 
 
-function createWindow() {
-  const win = new BrowserWindow({
-    width: 1024,
-    height: 768,
-  });
-  //win.maximize();
+async function createWindow() {
+ 
+  let freePort = null;
+  try {
+    freePort = await portfinder.getPortPromise({ port: 8001 });
+  } catch (err) {
+    console.error("Failed to obtain a port", err);
+    app.quit();
+  }
+  const serverUrl = `http://127.0.0.1:${freePort}/`;
 
   // Spawn the server process
   p = findAppPath()
-  const serverProcess = spawn(p)
+  const serverProcess = spawn(p, {env: {
+    SACRO_APP_TOKEN: RANDOM_SECRET,
+    PORT: freePort,
+  }})
+
+  // set a cookie with the random token in
+  const cookie = { url: serverUrl, name: 'sacro_app_token', value: RANDOM_SECRET }
+  session.defaultSession.cookies.set(cookie)
```

Comment:
Would you be better off `await`ing this and letting the uncaught error kill the app rather than logging and moving on? Would be less code as well...

Reply:
Hmm, maybe. I just c/p the example from the node docs, but will try awaiting